utils.go

```diff
@@ -269,18 +269,28 @@ func setupIO(process *libcontainer.Process, rootuid int, console string, createT
 	return createStdioPipes(process, rootuid)
 }
 
+// createPidFile creates a file with the processes pid inside it atomically
+// it creates a temp file with the paths filename + '.' infront of it
+// then renames the file
 func createPidFile(path string, process *libcontainer.Process) error {
 	pid, err := process.Pid()
 	if err != nil {
 		return err
 	}
-	f, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0666)
+	var (
+		tmpDir  = filepath.Dir(path)
+		tmpName = filepath.Join(tmpDir, fmt.Sprintf(".%s", filepath.Base(path)))
```

Comment:
I'm a bit late for this, I'm just wondering if there's a reason why TemFile(https://golang.org/pkg/io/ioutil/#TempFile) is not being used instead.